relational/table.py

```diff
@@ -770,31 +891,95 @@ def rows(self, proj):
         (predev_date, predev_data) = self._predevelopment_date(proj)
         (filed_date, filed_data) = self._filed_for_entitlements_date(proj)
         (entitled_date, entitled_data) = self._entitled_date(proj)
+        (permits_date, permits_data) = self._filed_for_permits(proj)
+        (construction_date, construction_data) = self._under_construction(proj)
+        (completed_date, completed_data) = self._completed_construction(proj)
 
         result = []
+        # Predevelopment doesn't necessarily happen for all projects
         if predev_date:
+            if filed_date and filed_date < predev_date:
+                print("Error: Project %s has pre-development date %s and "
```

Comment:
intentional to proceed even if this is an error? same elsewhere

because we have might have many thousands of entries, i'd prefer some sort of error sampling instead of overwhelming the terminal. might need an issue created to centralize all the sampled error messages.

Reply:
Updated to print out a random sampling. It was intentional to proceed even with an error since I thought it might still have value for there to be values in the status relational tables. However, does this make the Power BI side harder? If it's better on that side I can certainly just not continue in these cases?